text/0470-aurora-serverless-v2.md

```diff
@@ -0,0 +1,173 @@
+# Aurora Serverless v2 support
+
+* **Original Author(s):**: @pahud
+* **Tracking Issue**: #470
+* **API Bar Raiser**: @vinayak-kukreja
+
+Allow users to create Amazon Aurora Serverless v2 instances with the `aws-rds` module.
+
+## Working Backwards
+
+> This section should contain one or more "artifacts from the future", as if the
+> feature was already released and we are publishing its CHANGELOG, README,
+> CONTRIBUTING.md and optionally a PRESS RELEASE. This is the most important
+> section of your RFC. It's a powerful thought exercise which will challenge you
+> to truly think about this feature from a user's point of view.
+>
+> Choose *one or more* of the options below:
+>
+> * **CHANGELOG**: Write the changelog entry for this feature in conventional
+>   form (e.g. `feat(eks): cluster tags`). If this change includes a breaking
+>   change, include a `BREAKING CHANGE` clause with information on how to
+>   migrate. If migration is complicated, refer to a fictional GitHub issue and
+>   add its contents here.
+
+feat(rds): Aurora Serverless v2 support
+
+>
+> * **README**: If this is a new feature, write the README section which
+>   describes this new feature. It should describe the feature and walk users
+>   through usage examples and description of the various options and behavior.
+>
+
+## Aurora Serverless v2 support
+
+Aurora Serverless v2 is an on-demand, autoscaling configuration for Amazon Aurora.
+Aurora Serverless v2 helps to automate the processes of monitoring the workload and
+adjusting the capacity for your databases. Capacity is adjusted automatically based on
+application demand. Read
+[Using Aurora Serverless v2](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless-v2.html)
+for more details.
+
+### Create a new cluster
+
+Use `DatabaseClusterV2` to create a new cluster with Aurora Serverless V2 support.
+You may specify the `writer` and `readers` instances with the cluster. The instances
+could be either serverless or privisioned.
+
+```ts
+new rds.DatabaseClusterV2(stack, 'cluster', {
+  engine,
+  vpc,
+  // writer(serverless)
+  writer: { serverless: true },
```

Comment:
What does `serverless: true` mean here? A serverless instance has minimum and maximum ACUs right? What are those defaults for us? How can the customer modify them? Do we plan to add to the documentation about expenses this would add?

Reply:
The default ACU is defined in the `serverlessV2Scaling` prop for `DatabaseClusterV2` which is optional and the default is min: 0.5 max: 1, and that is the minimal required value. We are not allowed to define ACU for individual serverless instances. The only available configuration is the `serverlessV2Scaling` for the cluster.

Reply:
Hmm, could you please tell me why the maximum is 1? I see it can be higher [here](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless-v2.how-it-works.html#aurora-serverless-v2.how-it-works.capacity).

> We are not allowed to define ACU for individual serverless instances.

By this you mean the config of `min and max` that we would set would apply to all serverless v2 instances. Right?